From 99f00b72002404dade077e3fb388ab8570d31206 Mon Sep 17 00:00:00 2001
From: Ken Zhu <guigenz@codeaurora.org>
Date: Mon, 2 Apr 2018 17:53:13 -0700
Subject: [PATCH] map: change the way setting ipv6 address

It used to add a dynamic interface to set up the ipv6
address, that is unneccessary, it make the system more
complex, and cause the netifd crashed.

Set the IPv6 address directly to the tunnel interface.

Change-Id: I94915e1d4b69254486f9c25cc870a9f2183128d8
Signed-off-by: Ken Zhu <guigenz@codeaurora.org>
---
 package/network/ipv6/map/files/map.sh | 22 ++++++++--------------
 1 file changed, 8 insertions(+), 14 deletions(-)

diff --git a/package/network/ipv6/map/files/map.sh b/package/network/ipv6/map/files/map.sh
index 0115286b2e5d..df022ddac620 100755
--- a/package/network/ipv6/map/files/map.sh
+++ b/package/network/ipv6/map/files/map.sh
@@ -198,27 +198,21 @@ proto_map_setup() {
 			json_close_object
 		}
 		proto_add_ipv6_route $(eval "echo \$RULE_${k}_IPV6ADDR") 128
-	  fi
-	json_close_array
-	proto_close_data
 
-	proto_send_update "$cfg"
+	fi
 
 	if [ "$maptype" = "lw4o6" -o "$maptype" = "map-e" ]; then
-		json_init
-		json_add_string name "${cfg}_"
-		json_add_string ifname "@$(eval "echo \$RULE_${k}_PD6IFACE")"
-		json_add_string proto "static"
-		json_add_array ip6addr
 		if [ "$mode" = br ]; then
-			json_add_string "" "$(eval "echo \$RULE_${k}_BR")"
+			proto_add_ipv6_address "$(eval "echo \$RULE_${k}_BR")" "128"
 		else
-			json_add_string "" "$(eval "echo \$RULE_${k}_IPV6ADDR")"
+			proto_add_ipv6_address "$(eval "echo \$RULE_${k}_IPV6ADDR")" "128"
 		fi
-		json_close_array
-		json_close_object
-		ubus call network add_dynamic "$(json_dump)"
 	fi
+
+	json_close_array
+	proto_close_data
+
+	proto_send_update "$cfg"
 }
 
 proto_map_teardown() {
-- 
2.34.1

